great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -510,24 +453,9 @@ def test_connection(self) -> None:
                 f'"{self.table_name}" does not exist.'
             )
 
-    def add_year_and_month_splitter(
-        self,
-        column_name: str,
-    ) -> TableAsset:
-        """Associates a year month splitter with this TableAsset
-
-        Args:
-            column_name: A column name of the date column where year and month will be parsed out.
-
-        Returns:
-            This TableAsset so we can use this method fluently.
-        """
-        column_splitter = SqlYearMonthSplitter(
-            column_name=column_name,
-        )
-        column_splitter.test_connection(table_asset=self)
-        self.column_splitter = column_splitter
-        return self
+    def test_column_splitter_connection(self) -> None:
+        if self.column_splitter:
+            self.column_splitter.test_connection(table_asset=self)
```

Comment:
I don't see why it is necessary to refactor this out and create a new method name when right now we use `test_connection` across the board. In `Datasource.test_connection()` I added a parameter called `test_assets: bool` which can determine if the datasource assets should also be tested. Maybe we want to add `test_column_splitter: bool` as well? I feel like it should be necessary to first confirm your `DataAsset` connection is valid before you can ever test the splitter connection.

Reply:
[This](https://github.com/great-expectations/great_expectations/blob/ad95ef6d03a5302d5b4359ebe361537df39c3b9a/great_expectations/experimental/datasources/sql_datasource.py#L605) is what I'm referring to for reference. We can do it either way, but I like how clean it is to have the same method name on each class.

Reply:
On the data asset level, I feel like it should be testing its own splitter. Or we could make this static.